pdns/rec_channel_rec.cc

```diff
@@ -219,6 +219,47 @@ string static doGetParameter(T begin, T end)
   return ret;
 }
 
+/* Read an (open) fd from the control channel */
+static int
+getfd(int s)
```

Comment:
A small nit but I would really like to see that file descriptor wrapped in an object like our `Socket` object, to make sure it is properly closed when it goes out of scope. It looks like you are properly handling exceptions today, though, but it might not always be the case.